File: tests/sinus/sinus.py
def member_n(s_n_prev, x, member_prev, n_prev):
  factor = 2 * n_prev
  return (-1) * member_prev * x * x / (factor * (factor + 1))

def sin_series(arg, eps):
  s = arg
  n = 1
  x_n = arg
  while abs(x_n) > eps:
    x_n = member_n(s, arg, x_n, n)
    s = s + x_n
    # print("x_%d = %.4f, s=%.4f" %(n, x_n, s))
    n = n + 1
  return s

eps = 0.0001
print("epsilon = %f\n" %(eps))
arg = 17.0
sum = 0.0

for x in range(10000000):
  sum = sin_series(arg, eps)

print("Answer: sin(%f) = %f, eps = %f" %(arg, sum, eps))

File: tests/sinus/sinus.cpp
#include <iostream>
#include <cmath>

double member_n(double s_n_prev, double x, double member_prev, int n_prev) {
  int factor = 2 * n_prev;
  return (-1) * member_prev * x * x / (factor * (factor + 1));
}

double sin_series(double arg, double eps) {
  double s = arg;
  int n = 1;
  double x_n;
  for (x_n = arg; fabs(x_n) > eps; n = n + 1) {
    x_n = member_n(s, arg, x_n, n);
    s = s + x_n;
    //printf("x_%d = %.4f, s=%.4f", n, x_n, s);
  }
  return s;
}


int main() {
  double eps = 0.0001;
  printf("epsilon = %f\n", eps);
  double arg = 17;
  double sum;

  int i = 0;
  for (i = 0; i < 10000000; i = i+1) {
    sum = sin_series(arg, eps);
  }

  printf("Answer: sin(%f) = %f, eps = %f\n", arg, sum, eps);
}
